Mathlib/Data/Real/Pi/Irrational.lean

```diff
@@ -123,8 +123,7 @@ private lemma recursion (n : ℕ) :
     I (n + 2) θ * θ ^ 2 =
       2 * (n + 2) * (2 * n + 3) * I (n + 1) θ - 4 * (n + 2) * (n + 1) * I n θ := by
   rw [recursion' (n + 1)]
-  simp
-  ring!
```

Comment:
Isn't this a pattern that the flexible linter should allow?
Or otherwise, why is your suggested replacement more robust?

Reply:
@adomani Do you remember why the flexible linter was configured this way?

Reply:
I've reverted the change now - let's not get this PR held up on this.